worker-languagedetection/src/main/java/com/hpe/caf/worker/languagedetection/LanguageDetectionWorker.java

```diff
@@ -146,188 +132,30 @@ public void processDocument(Document document) throws InterruptedException, Docu
-            } else {
-                LOG.debug("Field value data is local.");
-                is = new ByteArrayInputStream(fieldValue.getValue());
-            }
-        } catch (DataStoreException dse) {
-            LOG.error("Failed to acquire source data from the remote data store");
-            // Convert to unchecked exception for streams api usage.
-            throw new RuntimeException(dse);
-        }
+        try (final SequenceInputStream sequenceInputStream = getFieldValuesAsStreams(sourceDataField, dataStore)) {
 
-        return is;
-    }
-
-    private static String getLanguageNameFieldName(int detectedLanguageId)
-    {
-        return LanguageDetectionConstants.Fields.DETECTED_LANGUAGE_PREFIX
-            + String.valueOf(detectedLanguageId)
-            + "_"
-            + LanguageDetectionConstants.Fields.DETECTED_LANGUAGE_NAME_SUFFIX;
-    }
-
-    private static String getLanguageCodeFieldName(int detectedLanguageId)
-    {
-        return LanguageDetectionConstants.Fields.DETECTED_LANGUAGE_PREFIX
-            + String.valueOf(detectedLanguageId)
-            + "_"
-            + LanguageDetectionConstants.Fields.DETECTED_LANGUAGE_CODE_SUFFIX;
-    }
-
-    private static String getLanguagePercentageFieldName(int detectedLanguageId)
-    {
-        return LanguageDetectionConstants.Fields.DETECTED_LANGUAGE_PREFIX
-            + String.valueOf(detectedLanguageId)
-            + "_"
-            + LanguageDetectionConstants.Fields.DETECTED_LANGUAGE_PERCENTAGE_SUFFIX;
-    }
-
-    private static void outputDocumentFieldValueChanges(final Document document)
-    {
-        final String baseOutputDir = System.getenv("CAF_LANG_DETECT_WORKER_OUTPUT_FOLDER");
-        final String outputSubdir = document.getCustomData("outputSubfolder");
-
-        // Only output document field value changes if configured to do so
-        if (baseOutputDir == null || baseOutputDir.isEmpty()) {
-            LOG.debug("No response data output folder specified.");
-            return;
-        }
-
-        LOG.debug("Outputting document field value changes.");
-
-        final Path outputFir = getFullOutputPath(baseOutputDir, outputSubdir);
-        final File outputFile = getFilepath(outputFir, document).toFile();
+            // Perform language detection.
+            LOG.debug("Perform language detection.");
+            final LanguageDetectorResult detectorResult = languageDetector.detectLanguage(sequenceInputStream);
 
-        // Iterate through each of the document fields and output changes where they exist.
-        document.getFields().forEach(field -> {
-            try {
-                appendFieldValueChangesToFile(field, outputFile);
-            } catch (IOException ioe) {
-                LOG.warn("Failed to output document field value changes", ioe);
+            if (detectorResult != null) {
+                addDetectedLanguageToDocument(detectorResult, document, sourceDataField, addFieldsInNewFormat);
```

Comment:
Need a more descriptive name than "addFieldsInNewFormat". Try to refer to new functionality by what it really represents, rather than simply by the fact that it is new.